pjsip/src/pjsip/sip_auth_client.c

```diff
@@ -291,7 +296,11 @@ PJ_DEF(void) pjsip_auth_create_digestSHA256(pj_str_t *result,
 
     } else if ((cred_info->data_type & PASSWD_MASK) == PJSIP_CRED_DATA_DIGEST)
     {
-	pj_assert(cred_info->data.slen == 32);
+	if (cred_info->data.slen != 32) {
+	    pj_bzero(result->ptr, result->slen);
+	    result->slen = 0;
+	}
+	PJ_ASSERT_RETURN(cred_info->data.slen == 32, PJ_EINVAL);
```

Comment:
same as above.
also perhaps the len of `data/ha1` should be 64 or `PJSIP_SHA256STRLEN` instead of 32 (need to check the standard)?